design/ceph/object/bucketpolicy.md

```diff
@@ -0,0 +1,93 @@
+  - _bucketPolicy_ in the `parameters` section of `StorageClass` for `ObjectBucketClaim`
+
+Policies need to provide in generic `json` format. A policy can have multiple `statement`.
+
+The following checks are needed to perform in Rook to verify whether the `bucketpolicy` applicable to OBC.
+- for ceph object user, `Principal` value should have username and `Resource` should have specific bucket names. It can be defined for buckets which are not part of the OBC as well, the `bucketname` of an OBC can be fetched from its `configmap`.
+- In `StorageClass`, `Principal` value should be `*`(applicable to all users) and `Resource` should have the bucket name can be empty since it can be generated name from Rook as well and will be attached before setting the policy.
+
+Examples:
+```yaml
+# storageclass-bucket.yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-ceph-delete-bucket
+provisioner: rook-ceph.ceph.rook.io/bucket
+# set the reclaim policy to delete the bucket and all objects
+# when its OBC is deleted.
+reclaimPolicy: Delete
+parameters:
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-east-1
+  bucketName: ceph-bkt
+  bucketPolicy: "Version": "2012-10-17",
+		"Statement": [{
+		"Effect": "Allow",
+		"Principal": {"AWS": ["arn:aws:iam:::*"]},
+		"Action": "s3:PutObjectAcl",
+		"Resource": [
+		"arn:aws:s3:::/*"
+		]
+	}]
+
+# object-user.yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStoreUser
+metadata:
+  name: my-user
+  namespace: rook-ceph
+spec:
+  store: my-store
+  displayName: "my display name"
+  bucketPolicy: "Version": "2012-10-17",
+		"Statement": [{
+		"Effect": "Allow",
+		"Principal": {"AWS": ["arn:aws:iam:::my-user"]},
+		"Action": "s3:PutObjectAcl",
+		"Resource": [
+		"arn:aws:s3:::ceph-bkt/*"
+		]
+	}]
+```
+
+The `bucketPolicy` defined on CRD won't override any existing policies on that bucket, will just append. But this can be easily overwritten with help of S3 client, since does not have much control over there.
+
+## Apis and structural changes
+
+The following field will be add to `ObjectStoreUserSpec` and this need to reflected on the existing api's for `CephObjectStoreUser`
+
+```
+type ObjectStoreUserSpec struct {
+        Store string `json:"store,omitempty"`
+        //The display name for the ceph users
+        DisplayName string `json:"displayName,omitempty"`
++       //The list of bucket policy for this user
++       BucketPolicy string `json:"bucketPolicy,omitempty"`
+ }
+```
+
+The `bucket policy` feature is consumed by the brownfield use case of `OBC`, so supporting apis and structures already exists in [policy.go](/pkg/operator/ceph/object/policy.go). Still few more api's are need to read the policy from CRD, validate it and then convert it into `bucketpolicy`, so that can be consumed by existing api's
+
+1. `func ReadBucketPolicy(CephObjectStoreUser.Spec/storagev1.StorageClass) string {}`
```

Comment:
No need to mention method names in the design doc. These could change over time, then the design doc would be outdated.